entities/entities/food/init.lua

```diff
@@ -25,19 +25,27 @@ function ENT:OnTakeDamage(dmg)
 		effectdata:SetScale(2)
 		effectdata:SetRadius(3)
 		util.Effect("Sparks", effectdata)
-		self:Remove()
+		SafeRemoveEntity(self)
 	end
 end
 
 function ENT:Use(activator,caller)
-	caller:setSelfDarkRPVar("Energy", math.Clamp((caller:getDarkRPVar("Energy") or 0) + 100, 0, 100))
+	local override = self.foodItem and self.foodItem.onEaten and self.foodItem.onEaten(self, activator, self.foodItem) or isfunction(self.onEaten) and self.onEaten(self, activator, self.foodItem)
+
+	if override then
+		SafeRemoveEntity(self)
+		return
+	end
+	
+	caller:setSelfDarkRPVar("Energy", GAMEMODE.Config.maxhunger)
```

Comment:
Why should the "max hunger" ever by anything other than 100? You already have hungerspeed.